broker/src/main/java/org/apache/rocketmq/broker/transaction/queue/TransactionalMessageServiceImpl.java

```diff
@@ -431,7 +431,7 @@ private MessageQueue getOpQueue(MessageQueue messageQueue) {
         if (opQueue == null) {
             opQueue = new MessageQueue(TransactionalMessageUtil.buildOpTopic(), messageQueue.getBrokerName(),
                 messageQueue.getQueueId());
-            opQueueMap.put(messageQueue, opQueue);
+            opQueueMap.putIfAbsent(messageQueue, opQueue);
```

Comment:
It seems that if `opQueueMap` contains `messageQueue`, the newly created `opQueue` would not be put in `opQueueMap` but returned.

Reply:
Thank you for reminding me